### PR TITLE
Merge ProductPrice atomic block and component

### PR DESCRIPTION
**Pull request by Aljullu**
Part of #1967.

This PR:
* Updates the `ProductPrice` component so it can handle price ranges and loading states, in order to have feature-parity with the Product Price atomic block.
* Moves the `ProductPrice` component out of the `cart-checkout` subdirectory.
* Updates the Product Price atomic block so internally it uses the `ProductPrice` component, some benefits of this:
  * The markup is centralized in a single place. So any enhancement we do, will affect Cart & Checkout + All Products & Single Product.
  * Similarly, styles can all be in a single file too. Preventing things like #3048.

### How to test the changes in this Pull Request:

Testing basically means verifying there are no regressions:

1. Load the Cart and Checkout blocks and verify product prices (default prices and on sale prices) are still rendered correctly.

2. Create a page with the Product Price block for three products (one with a default price, another one on sale and another one with a price range--grouped product). Verify prices display correctly and you can style them in the editor.

3. Add the All Products block in a page and verify there are no regressions with the Product Price block.

### CODE REVIEW

File: assets/js/base/components/cart-checkout/product-name/index.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { decodeEntities } from '@wordpress/html-entities';
  */
 import './style.scss';
 
-const ProductName = ( { name, permalink, disabled } ) => {
+const ProductName = ( { name, permalink, disabled = false } ) => {
```
Comment (Aljullu) [inline assets/js/base/components/cart-checkout/product-name/index.js:12]: Changes in this file are not related to this PR, but linting was complaining about it.

**Review by nerrad — APPROVED** [review]
This is great work for consolidating components here @Aljullu.  I've left a couple minor notes that you can consider.

I haven't tested this but code-wise this looks okay.

It'd be nice to have a storybook for these components.

File: assets/js/atomic/blocks/product-elements/price/block.js [hunk 2]
```diff
@@ -80,163 +83,39 @@ const Block = ( {
...
+			align={ align }
+			className={ wrapperClassName }
+			currency={ currency }
+			price={ prices.price }
+			priceClassName={ priceClassName }
+			priceStyle={ isFeaturePluginBuild() ? priceStyle : {} }
+			// Range price props
+			minPrice={ prices.price_range && prices.price_range.min_amount }
+			maxPrice={ prices.price_range && prices.price_range.max_amount }
+			// On sale price props
```
[the remark this answers is not in the document]
Comment (Aljullu) [reply assets/js/atomic/blocks/product-elements/price/block.js:111]: Yeah, agree those names are quite confusing, I think they are based on the internal WC names. For now, I updated the comment with your suggestion.

> I'm guessing it's something like this:
> * regularPrice is present when the actual price is a sale price.
> * price is the actual price for the item (which may be a sale price).
> Is that correct?

Yes, that's my understanding as well.

**Comment by Aljullu** [comment]
Thanks for the review @nerrad!

> It'd be nice to have a storybook for these components.

Added in 669360ac53800c9144612e5949d1689be970ec4e. :slightly_smiling_face:

**Comment by Aljullu** [comment]
It looks like #2991 and #3080 have been fixed in this PR, I added a changelog notice accordingly.

**Pull request merged**